[bug] AutotoolsDeps : LIBS doesn't include system_libs

from https://github.com/conan-io/conan-center-index/pull/9488
working on the recipe, I found that test package failed:
```
Scanning dependencies of target test_package
[ 50%] Building C object CMakeFiles/test_package.dir/test_package.c.o
[100%] Linking C executable bin/test_package
/usr/bin/ld: bin/test_package: hidden symbol `pthread_atfork' in /usr/lib/x86_64-linux-gnu/libpthread_nonshared.a(pthread_atfork.oS) is referenced by DSO
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
CMakeFiles/test_package.dir/build.make:108: recipe for target 'bin/test_package' failed
make[2]: *** [bin/test_package] Error 1
CMakeFiles/Makefile2:94: recipe for target 'CMakeFiles/test_package.dir/all' failed
make[1]: *** [CMakeFiles/test_package.dir/all] Error 2
Makefile:102: recipe for target 'all' failed
make: *** [all] Error 2
```
simple workaround in recipe fixes the issue:
```
@@ -61,6 +62,7 @@ class KModConan(ConanFile):
         tc.generate()
         tc = AutotoolsDeps(self)
         tc.environment.define("PKG_CONFIG_PATH", self.source_folder)  # is it really needed?
+        tc.environment.define("LIBS", "-lpthread")
         tc.generate()

     def requirements(self):
```
I've compared the logs, and found, one that doesn't work:
```
348 libtool: link: /usr/bin/gcc -shared  -fPIC -DPIC  libkmod/.libs/libkmod.o libkmod/.libs/libkmod-builtin.o libkmod/.libs/libkmod-list.o libkmod/.libs/libkmod-config.o libkmod/.libs/libkmod-index.o libkmod/.libs/libkmod-module.o libkm    od/.libs/libkmod-file.o libkmod/.libs/libkmod-elf.o libkmod/.libs/libkmod-signature.o  -Wl,--whole-archive shared/.libs/libshared.a -Wl,--no-whole-archive  -L/home/conan/.conan/data/zstd/1.5.2/_/_/package/ff7e1eda45846fc95249ac3e0c5    0da0e296272b3/lib -L/home/conan/.conan/data/xz_utils/5.2.5/_/_/package/76f87539fc90ff313e0b3182641a9bb558a717d2/lib -L/home/conan/.conan/data/zlib/1.2.11/_/_/package/76f87539fc90ff313e0b3182641a9bb558a717d2/lib -L/home/conan/.conan/    data/openssl/3.0.1/_/_/package/d5204605bf864d4bd98fc6a9e624938e8e04ad03/lib -ldl -lrt -lpthread -lzstd -llzma -lz -lssl -lcrypto  -m64 -O3 -Wl,--version-script=/home/conan/.conan/data/kmod/29/_/_/build/e7842461bbcd29441fdb340a5299c0    19eae366ea/source_subfolder/libkmod/libkmod.sym -m64   -Wl,-soname -Wl,libkmod.so.2 -o libkmod/.libs/libkmod.so.2.3.7
```
one that works:
```
348 libtool: link: /usr/bin/gcc -shared  -fPIC -DPIC  libkmod/.libs/libkmod.o libkmod/.libs/libkmod-builtin.o libkmod/.libs/libkmod-list.o libkmod/.libs/libkmod-config.o libkmod/.libs/libkmod-index.o libkmod/.libs/libkmod-module.o libkm    od/.libs/libkmod-file.o libkmod/.libs/libkmod-elf.o libkmod/.libs/libkmod-signature.o  -Wl,--whole-archive shared/.libs/libshared.a -Wl,--no-whole-archive  -L/home/conan/.conan/data/zstd/1.5.2/_/_/package/ff7e1eda45846fc95249ac3e0c5    0da0e296272b3/lib -L/home/conan/.conan/data/xz_utils/5.2.5/_/_/package/76f87539fc90ff313e0b3182641a9bb558a717d2/lib -L/home/conan/.conan/data/zlib/1.2.11/_/_/package/76f87539fc90ff313e0b3182641a9bb558a717d2/lib -L/home/conan/.conan/    data/openssl/3.0.1/_/_/package/d5204605bf864d4bd98fc6a9e624938e8e04ad03/lib -lzstd -llzma -lcrypto -ldl -lrt -lz -lpthread  -m64 -O3 -Wl,--as-needed -Wl,--no-undefined -Wl,--gc-sections -Wl,--version-script=/home/conan/.conan/data/k    mod/29/_/_/build/e7842461bbcd29441fdb340a5299c019eae366ea/source_subfolder/libkmod/libkmod.sym -m64   -Wl,-soname -Wl,libkmod.so.2 -o libkmod/.libs/libkmod.so.2.3.7
```

so, that's the essential difference?
taking a look at `config.status` I see that:
```
S["libcrypto_LIBS"]="-L/home/conan/.conan/data/openssl/3.0.1/_/_/package/d5204605bf864d4bd98fc6a9e624938e8e04ad03/lib -L/home/conan/.conan/data/zlib/1.2.11/_/_/package/7"\
S["zlib_LIBS"]="-L/home/conan/.conan/data/zlib/1.2.11/_/_/package/76f87539fc90ff313e0b3182641a9bb558a717d2/lib -lz -Wl,-rpath,/home/conan/.conan/data/zlib/1.2.11/_/"\
S["liblzma_LIBS"]="-L/home/conan/.conan/data/xz_utils/5.2.5/_/_/package/76f87539fc90ff313e0b3182641a9bb558a717d2/lib -llzma -lpthread -Wl,-rpath,/home/conan/.conan/dat"\
S["libzstd_LIBS"]="-L/home/conan/.conan/data/zstd/1.5.2/_/_/package/ff7e1eda45846fc95249ac3e0c50da0e296272b3/lib -lzstd -lpthread -Wl,-rpath,/home/conan/.conan/data/zs"\
S["LIBS"]=" -lzstd -llzma -lz -lssl -lcrypto"
```
in other words, `LIBS` doesn't include `system_libs`, and the variable came from `AutotoolsDeps`.



<!--
  Please don't forget to update the issue title.
  Include all applicable information to help us reproduce your problem.

  To help us debug your issue please explain:
-->

### Environment Details (include every applicable attribute)
  * Operating System+version:
  * Compiler+version:
  * Conan version:
  * Python version:

### Steps to reproduce (Include if Applicable)


### Logs (Executed commands with output) (Include/Attach if Applicable)

<!--
  Your log content should be related to the bug description, it can be:
  - Conan command output
  - Server output (Artifactory, conan_server)
-->

--- a/conan/tools/gnu/autotoolsdeps.py
+++ b/conan/tools/gnu/autotoolsdeps.py
@@ -37,6 +37,10 @@ class AutotoolsDeps:
             ldflags.extend(flags.framework_paths)
             ldflags.extend(flags.lib_paths)
 
+            # libs
+            libs = flags.libs
+            libs.extend(flags.system_libs)
+
             # cflags
             cflags = flags.cflags
             cxxflags = flags.cxxflags
@@ -49,7 +53,7 @@ class AutotoolsDeps:
 
             env = Environment()
             env.append("CPPFLAGS", cpp_flags)
-            env.append("LIBS", flags.libs)
+            env.append("LIBS", libs)
             env.append("LDFLAGS", ldflags)
             env.append("CXXFLAGS", cxxflags)
             env.append("CFLAGS", cflags)

--- a/conan/tools/gnu/gnudeps_flags.py
+++ b/conan/tools/gnu/gnudeps_flags.py
@@ -2,6 +2,7 @@
     This is a helper class which offers a lot of useful methods and attributes
 """
 # FIXME: only for tools.gnu? perhaps it should be a global module
+from conan.tools.microsoft import is_msvc
 from conan.tools.microsoft.subsystems import subsystem_path, deduce_subsystem
 
 
@@ -25,7 +26,7 @@ class GnuDepsFlags(object):
         self.cflags = cpp_info.cflags or []
         self.sharedlinkflags = cpp_info.sharedlinkflags or []
         self.exelinkflags = cpp_info.exelinkflags or []
-        self.system_libs = cpp_info.system_libs or []
+        self.system_libs = self._format_libraries(cpp_info.system_libs)
 
         # Not used?
         # self.bin_paths
@@ -47,8 +48,7 @@ class GnuDepsFlags(object):
             return []
         # FIXME: Missing support for subsystems
         compiler = self._conanfile.settings.get_safe("compiler")
-        compiler_base = self._conanfile.settings.get_safe("compiler.base")
-        if (str(compiler) not in self._GCC_LIKE) and (str(compiler_base) not in self._GCC_LIKE):
+        if str(compiler) not in self._GCC_LIKE:
             return []
         return ["-framework %s" % framework for framework in frameworks]
 
@@ -60,15 +60,13 @@ class GnuDepsFlags(object):
         if not framework_paths:
             return []
         compiler = self._conanfile.settings.get_safe("compiler")
-        compiler_base = self._conanfile.settings.get_safe("compiler.base")
-        if (str(compiler) not in self._GCC_LIKE) and (str(compiler_base) not in self._GCC_LIKE):
+        if str(compiler) not in self._GCC_LIKE:
             return []
-        return ["-F %s" % self._adjust_path(framework_path) for framework_path in
-                framework_paths]
+        return ["-F %s" % self._adjust_path(framework_path) for framework_path in framework_paths]
 
     def _sysroot_flag(self, sysroot):
         # FIXME: Missing support for subsystems
-        if self._base_compiler != 'Visual Studio' and sysroot:
+        if not is_msvc(self._conanfile) and sysroot:
             sysroot = self._adjust_path(sysroot)
             return '--sysroot=%s' % sysroot
         return ""
@@ -84,7 +82,7 @@ class GnuDepsFlags(object):
         if not library_paths:
             return []
         # FIXME: Missing support for subsystems
-        pattern = "-LIBPATH:%s" if self._base_compiler == 'Visual Studio' else "-L%s"
+        pattern = "-LIBPATH:%s" if is_msvc(self._conanfile) else "-L%s"
         return [pattern % self._adjust_path(library_path)
                 for library_path in library_paths if library_path]
 
@@ -93,10 +91,8 @@ class GnuDepsFlags(object):
             return []
 
         result = []
-        compiler = self._conanfile.settings.get_safe("compiler")
-        compiler_base = self._conanfile.settings.get_safe("compiler.base")
         for library in libraries:
-            if str(compiler) == 'Visual Studio' or str(compiler_base) == 'Visual Studio':
+            if is_msvc(self._conanfile):
                 if not library.endswith(".lib"):
                     library += ".lib"
                 result.append(library)
@@ -105,15 +101,10 @@ class GnuDepsFlags(object):
         return result
 
     def _adjust_path(self, path):
-        if self._base_compiler == 'Visual Studio':
+        if is_msvc(self._conanfile):
             path = path.replace('/', '\\')
         else:
             path = path.replace('\\', '/')
 
         path = subsystem_path(self._subsystem, path)
         return '"%s"' % path if ' ' in path else path
-
-    @property
-    def _base_compiler(self):
-        return str(self._conanfile.settings.get_safe("compiler.base") or
-                   self._conanfile.settings.get_safe("compiler"))

--- a/conans/test/unittests/client/toolchain/autotools/autotools_deps_test.py
+++ b/conans/test/unittests/client/toolchain/autotools/autotools_deps_test.py
@@ -95,6 +95,10 @@ def test_foo():
 
         assert env["CXXFLAGS"] == 'dep1_a_cxx_flag dep2_a_cxx_flag --sysroot=/path/to/folder/dep1'
         assert env["CFLAGS"] == 'dep1_a_c_flag dep2_a_c_flag --sysroot=/path/to/folder/dep1'
+        assert env["LIBS"] == "-ldep1_onelib -ldep1_twolib -ldep2_onelib -ldep2_twolib "\
+                              "-ldep1_onesystemlib -ldep1_twosystemlib "\
+                              "-ldep2_onesystemlib -ldep2_twosystemlib"
+
         folder = temp_folder()
         consumer.folders.set_base_install(folder)
         deps.generate()

